### PR TITLE
[ENH] Add threshold argument to make_adaptive_mask

### PR DESCRIPTION
**Pull request by tsalo**
References #617.

Changes proposed in this pull request:

- Add threshold argument to `make_adaptive_mask`.
- Hardcode threshold at 1 for the t2smap workflow and 3 for the tedana workflow.
- When deriving the binary mask from the adaptive mask, change threshold from 3 to 0 (basically assuming that the adaptive mask is thresholded).
- Improve associated documentation a bit.
- Update the `make_adaptive_mask` test.

### CODE REVIEW

**Review by jbteves — CHANGES_REQUESTED** [review]
The code looks good! However, I think the existing description of
> This mask may be thresholded, typically with values less than 3 set to 0 depending
        on the thresholding method.

Is a tad confusing, and it's in many places. I'm not sure the precise best way to address this, but maybe through either a piece of API documentation which each docstring can reference?

**Comment by tsalo** [comment]
I could include a "see also" to the `make_adaptive_mask` function in the docstrings of functions that use the adaptive mask. WDYT?

**Comment by jbteves** [comment]
That's perfect.

**Review by jbteves — APPROVED** [review]
LGTM; thanks @tsalo !

File: tedana/utils.py [hunk 1]
```diff
@@ -90,19 +93,20 @@ def make_adaptive_mask(data, mask=None, getsum=False):
...
-        mask = (masksum >= 3).astype(bool)
         # reduce mask based on masksum
         # TODO: Use visual report to make checking the reduced mask easier
-        if np.any(masksum[mask] < 3):
-            n_bad_voxels = np.sum(masksum[mask] < 3)
+        if np.any(masksum[mask] < threshold):
+            n_bad_voxels = np.sum(masksum[mask] < threshold)
             LGR.warning('{0} voxels in user-defined mask do not have good '
                         'signal. Removing voxels from mask.'.format(n_bad_voxels))
+            masksum[masksum < threshold] = 0
```
Comment (notZaki) [inline tedana/utils.py:109]: I am not completely familiar with where the adaptive mask is used, but could adding the following line
```python
masksum[maskum == 1] = 2
```
simplify the downstream steps or would it break other parts of the code?

Comment (tsalo) [reply tedana/utils.py:109]: I think it depends on where we want to bury the "treat one good echo as two good echoes" trick. The way it is now, we can dig into where the adaptive mask indicates only one good echo, and it's on the other functions to document that those voxels will be treated as if they have two good echoes (depending on the threshold). On the other hand, if we add that, then the other functions can be simplified a bit, but we won't be able to investigate the adaptive mask as well.

Comment (notZaki) [reply tedana/utils.py:109]: That makes sense. Thanks.

File: tedana/combine.py [hunk 2]
```diff
@@ -137,7 +136,9 @@ def make_optcom(data, tes, adaptive_mask, t2s=None, combmode='t2s', verbose=True
     tes : (E,) :obj:`numpy.ndarray`
         Array of TEs, in seconds.
     adaptive_mask : (S,) :obj:`numpy.ndarray`
-        Adaptive mask of the data indicating the number of echos with signal at each voxel
+        Adaptive mask of the data indicating the number of echos with signal at each voxel.
+        This mask may be thresholded, typically with values less than 3 set to 0 depending
```
Comment (emdupre) [inline tedana/combine.py:140]: Docstring suggestion for clarity :point_down: 

```suggestion
        This mask may be thresholded; for example, with values less than 3 set to 0.
        For more information on thresholding, see `make_adaptive_mask`.
```

File: tedana/decay.py [hunk 3]
```diff
@@ -73,7 +73,8 @@ def fit_monoexponential(data_cat, echo_times, adaptive_mask, report=True):
         Echo times in milliseconds.
     adaptive_mask : (S,) :obj:`numpy.ndarray`
         Array where each value indicates the number of echoes with good signal
-        for that voxel.
+        for that voxel. This mask may be thresholded, typically with values
```
Comment (emdupre) [inline tedana/decay.py:76]: Same comment as above re: docstring suggestion. Won't comment on other instances !

**Review by emdupre — DISMISSED** [review]
Thanks, @tsalo ! This approach makes sense to me. I made a light suggestion on the docstring, but not a strong feeling.

I think this will definitely help improve the T2*map use case !!

**Review by jbteves — DISMISSED** [review]

**Comment by tsalo** [comment]
Thanks @emdupre! I'm going to count both approvals and merge once tests pass.

**Pull request merged**